mindmeld/components/_config.py

```diff
@@ -498,6 +496,40 @@
     "allowed_patterns": [],
 }
 
+DEFAULT_ENTITY_RESOLVER_MODEL_CONFIGS = {
```

Comment:
I think its better to have a single config named `DEFAULT_ENTITY_RESOLVER_MODEL_CONFIG`, where the developer can then instantiate the following:

```
{
        'model_type': 'exact_match',
        "model_settings": {
            "augment_lower_case": False
}
```
or the other types of resolver matching. So there will be only two keys `model_type ` and `model_settings `. Moreover, I think it makes sense to rename `model_type ` to `type` and `model_settings ` to `params`, to be consistent with the other configs in mindmeld.

Reply:
As discussed, individualized and added the defaults as getter methods for classes that require it. From my understanding through [Mindmeld docs](https://www.mindmeld.com/docs/userguide/domain_classifier.html#classifier-configuration), the names are already consistent with other classifiers, right? Please correct me if you meant something else.

Reply:
Hey @vijay120, I gave some thought around this and looked at how the domain/intent/entity-recognition models are implemented. Going forward, I think it would be meaningful to have entity resolver models in the format similar to `TextModel` and `TaggerModel`. So the `'model_type'` should be same for all resolvers, something like `'model_type': 'matcher'` or `'model_type': 'resolver'`. And in the `'model_setting'`, we need to mention details such as 

```python
'model_setting': {
       'type': 'sbert_cosine_similarity', 
       'pretrained_name_or_abspath': 'distilbert-base-nli-stsb-mean-tokens', 
       ...
}
```
And also, we need to have add another file as `mindmeld/models/matching_models.py` wherein these matcher models are registered in `mindmeld/models/helpers.py`. I think this refactoring can bring resolver models in same format as other models. And this type of refactoring is probably required if we plan to include resolver models that can be trained/evaluated. 

What are your comments on this? I think it would be better to cover this refactoring in another PR. Also some previous refactoring comments from Marvin such as [this](https://github.com/cisco/mindmeld/pull/278#discussion_r592826409) can be included in that PR.